backend/models/loginRecord.js

```diff
@@ -0,0 +1,134 @@
+/**
+ *	Copyright (C) 2021 3D Repo Ltd
+ *
+ *	This program is free software: you can redistribute it and/or modify
+ *	it under the terms of the GNU Affero General Public License as
+ *	published by the Free Software Foundation, either version 3 of the
+ *	License, or (at your option) any later version.
+ *
+ *	This program is distributed in the hope that it will be useful,
+ *	but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *	GNU Affero General Public License for more details.
+ *
+ *	You should have received a copy of the GNU Affero General Public License
+ *	along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+"use strict";
+const db = require("../handler/db");
+
+// detects edge as browser but not device
+const UaParserJs = require("ua-parser-js");
+const Device = require("device");
+const Ip2location = require("ip-to-location");
+const Elastic = require("../handler/elastic");
+
+const LoginRecord = {};
+
+LoginRecord.saveLoginRecord = async (req) => {
```

Comment:
Actually can we pass in the user agent and ip here only?

If we can help it I want to avoid req being accessible anywhere within models folder.

Reply:
we use some other properties of req like sessionId, referrer and body.username. Shall I pass all the ones we use as params ?

Reply:
yep